### ISSUE TITLE
Instability of results when tested individually after model training.

### ISSUE THREAD

**Issue by xiaozheng-liu** [yangyangyang127/Seg-NN#1]
Hi yang, thank you for your excellent work！
I have some questions when debugging the code. There may be a problem with my reproduction steps. I hope you can help me.
We noticed that you integrated the training and testing code into one file and optimized the code structure.
We can get better results in the test set at the end of training, but when I retest using the trained model weights, the effect is significantly reduced. 
As you can see from the screenshot, the best result during training is 0.596, and the result when tested alone is 0.437, and each test will produce different results. What is the reason for this?
We hope to receive your reply, thank you. Thanks again for your excellent work!
![test1](https://github.com/yangyangyang127/TFS3D/assets/56385374/9cf5f986-0766-448f-a5f4-25ec4c578943)
![test2](https://github.com/yangyangyang127/TFS3D/assets/56385374/1249459c-78ca-4aaf-8beb-c61b76ec22cb)

**Comment by xiaozheng-liu** [comment]
Supplement: We only commented out the training part of the traning.py file
![1](https://github.com/yangyangyang127/TFS3D/assets/56385374/7f8abebc-4b0a-4792-9492-2e9ea7fadd33)
![2](https://github.com/yangyangyang127/TFS3D/assets/56385374/aab964f0-3651-4952-b733-23d9e9e6b05d)

**Comment by mooncakehub** [comment]
so if we use the code ,we just can train,but we can't test?

**Comment by xiaozheng-liu** [comment]
> so if we use the code ,we just can train,but we can't test?

Thanks for your reply. As I described above, testing is done at the end of the training phase and the results are good. However, when the weights are loaded separately for testing, the previous test results cannot be reproduced. And the results of each test are different. We don’t know exactly what the reason is. Have you ever encountered this problem? We hope you can answer it.

**Comment by xiaozheng-liu** [comment]
> so if we use the code ,we just can train,but we can't test?

Sorry, I mistakenly thought you were the author. To answer your question, yes, I was unable to reproduce the results during individual testing.

**Comment by mooncakehub** [comment]
I am interested in this code too,but if this code is incomplete, it is difficult for us to take advantage of it

**Comment by mooncakehub** [comment]
Because I don't have enough memory on my single gpu, I can't reproduce the code right now

**Comment by mooncakehub** [comment]
Because I don't have enough memory on my single gpu, I can't reproduce the code right now

**Comment by xiaozheng-liu** [comment]
> I am interested in this code too,but if this code is incomplete, it is difficult for us to take advantage of it

Yes, I can't be sure what the reason is, and I very much hope the author can help.

**Comment by xiaozheng-liu** [comment]
> Because I don't have enough memory on my single gpu, I can't reproduce the code right now

The code does not require much GPU memory, I only need about 5.4 G GPU memory during training.

**Comment by xiaozheng-liu** [comment]
> > Because I don't have enough memory on my single gpu, I can't reproduce the code right now
> 
> The code does not require much GPU memory, I only need about 5.4 G GPU memory during training.

What is your graphics card model?

**Comment by yangyangyang127** [comment]
> > so if we use the code ,we just can train,but we can't test?
> 
> Thanks for your reply. As I described above, testing is done at the end of the training phase and the results are good. However, when the weights are loaded separately for testing, the previous test results cannot be reproduced. And the results of each test are different. We don’t know exactly what the reason is. Have you ever encountered this problem? We hope you can answer it.

Sorry for the inconvenience the code caused. I have encountered issues with unstable results. But I do not know the problem you described. I will solve this problem in the next version.

**Comment by xiaozheng-liu** [comment]
> > > so if we use the code ,we just can train,but we can't test?
> > 
> > 
> > Thanks for your reply. As I described above, testing is done at the end of the training phase and the results are good. However, when the weights are loaded separately for testing, the previous test results cannot be reproduced. And the results of each test are different. We don’t know exactly what the reason is. Have you ever encountered this problem? We hope you can answer it.
> 
> Sorry for the inconvenience the code caused. I have encountered issues with unstable results. But I do not know the problem you described. I will solve this problem in the next version.

Thank you very much for your answer and we look forward to your improvements to the project！
